server/command.go

```diff
@@ -477,19 +476,7 @@ func (p *Plugin) executeConnectCommand(args *model.CommandArgs) (*model.CommandR
 		}
 	}
 
-	state := fmt.Sprintf("%s_%s", model.NewId(), args.UserId)
-	if err := p.store.StoreOAuth2State(state); err != nil {
-		p.API.LogWarn("Error in storing the OAuth state", "error", err.Error())
-		return p.cmdError(args.UserId, args.ChannelId, genericErrorMessage)
-	}
-
-	codeVerifier := model.NewId()
-	if appErr := p.API.KVSet("_code_verifier_"+args.UserId, []byte(codeVerifier)); appErr != nil {
-		p.API.LogWarn("Error in storing the code verifier", "error", appErr.Error())
-		return p.cmdError(args.UserId, args.ChannelId, genericErrorMessage)
-	}
-
-	connectURL := msteams.GetAuthURL(p.GetURL()+"/oauth-redirect", p.configuration.TenantID, p.configuration.ClientID, p.configuration.ClientSecret, state, codeVerifier)
+	connectURL := p.GetURL() + "/connect"
```

Comment:
Why is the PKCE part of the OAuth2 flow removed?

Reply:
It's not removed, just centralized. It happens in the connect api, [here](https://github.com/mattermost/mattermost-plugin-msteams/blob/a853fe81c8eef9e1ffb29143f8a17d30fef049ff/server/api.go#L316)

Rather than creating the redirect link in multiple places, we now create a `/connect` link which itself creates and handles the redirect.